backend/danswer/server/manage/users.py

```diff
@@ -69,11 +77,41 @@ async def demote_admin(
 
 @router.get("/manage/users")
 def list_all_users(
+    q: str,
+    accepted_page: int,
+    invited_page: int,
     _: User | None = Depends(current_admin_user),
     db_session: Session = Depends(get_session),
-) -> list[UserRead]:
-    users = list_users(db_session)
-    return [UserRead.from_orm(user) for user in users]
+) -> AllUsersResponse:
+    users = list_users(db_session, q=q)
+    accepted_emails = {user.email for user in users}
+    invited_emails = get_invited_users()
+    if q:
+        invited_emails = [
+            email for email in invited_emails if re.search(r"{}".format(q), email, re.I)
+        ]
+
+    accepted_count = len(accepted_emails)
+    invited_count = len(invited_emails)
+
+    return AllUsersResponse(
+        accepted=[
+            FullUserSnapshot(
+                id=user.id,
+                email=user.email,
+                role=user.role,
+                status=UserStatus.LIVE if user.is_active else UserStatus.BLOCKED,
+            )
+            for user in users
+        ][accepted_page * USERS_PAGE_SIZE : (accepted_page + 1) * USERS_PAGE_SIZE],
+        invited=[
+            InvitedUserSnapshot(email=email)
+            for email in invited_emails
+            if email not in accepted_emails
+        ][invited_page * USERS_PAGE_SIZE : (invited_page + 1) * USERS_PAGE_SIZE],
+        accepted_pages=accepted_count // USERS_PAGE_SIZE,
```

Comment:
nit: should this be `accepted_count // USERS_PAGE_SIZE + 1`?

Reply:
Yes!  Beautiful catch.

Reply:
Done: 870fc7d6